src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -383,6 +389,28 @@ bool ProcessPrimaryQueue ()
 			return true;
 		}
 
+		void ProcessPendingTypeChecks ()
+		{
+			for (int i = 0; i < pending_isinst_instr.Count; ++i) {
+				var item = pending_isinst_instr[i];
+				TypeDefinition type = item.Type;
+				if (Annotations.IsInstantiated (type))
+					continue;
+
+				Instruction instr = item.Instr;
+				Instruction prev_instr = instr.Previous;
+				ILProcessor ilProcessor = item.Body.GetILProcessor ();
+
+				if (prev_instr.OpCode.StackBehaviourPop == StackBehaviour.Pop0 && prev_instr.OpCode.StackBehaviourPush == StackBehaviour.Push1) {
```

Comment:
ldsfld seems to be the only one in this bucket and only if triggers static ctor for the first time and that would be observable only if the program was deterministic. Kind of tempted not to handle that.

Reply:
Compilers generally care about this sort of side effects because if a customer actually hits this, this is the sort of bug that will take a lot of someone's time. IL Linker is a compiler tool.

E.g.

```csharp
public class C {
    int x;
    int y;
    public void M(C c) {
        _ = c.x + c.y;
    }
}
```

Compiles to

```fasm
C.M(C)
    L0000: mov eax, [edx+4]
    L0003: ret
```

(I.e. RyuJIT knows that the reads are dead and eliminates the addition, but it keeps at least one read in the method body to trigger a NullReferenceException if necessary.)

Reply:
Either way, I don't think the code that is there right now will work as-is because we're not checking whether the previous instruction belongs to the same basic block. I'm not sure the savings of one instruction are worth all the trouble with it.

I believe this will end up generating invalid IL for:

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAmARgFgAoHAgNgAJ8qBhKgb1KtZsqoFEA7AVzCoBZABQB7YACs4AYxhVRBJPMky5ovAEomLNrpwB2KmIJUA/Kfmaq0LnzABuHawC+pZ0A===

Reply:
Good point about the jump reachability. Removed the logic for now.